addons/stock_picking_batch/models/stock_move_line.py

```diff
@@ -0,0 +1,9 @@
+# -*- encoding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import fields, models
+
+class StockMoveLine(models.Model):
+    _inherit = 'stock.move.line'
+
+    batch_id = fields.Many2one(related='picking_id.batch_id')
```

Comment:
i'd remove that with the two related i suggested earlier

Reply:
I removed it, and I added the file into enterprise (`stock_barcode_picking_batch/models/stock_move_line.py`) to add a domain on `picking_id` field in the move line *"Add a line"* form view.
I need this domain because from a batch, the user must manually define to which picking the move line is linked to.